### ISSUE TITLE
Use Ygg to provide shared libraries from Yggdrasil outside of Julia

### ISSUE THREAD

**Issue by amontoison** [fredrikekre/ygg#11]
@fredrikekre 
Is it possible that `Ygg` install shared libraries from Yggdrasil in the future such that we can use them outside of Julia?
An example of artifact that we would like to use outside Julia is the shared library of `METIS4_jll`.

**Comment by fredrikekre** [comment]
It is not possible right now, but perhaps it could be implemented.

Just to clarify, you want to e.g. be able to pass something to `-I` and `-L` when invoking `gcc` or something? That should be doable by mimicking what BinaryBuilder does, which just unpacks all dependencies into a common prefix, right @giordano? Or I guess you symlink everything from a common prefix to the actual artifact pahts?

**Comment by fredrikekre** [comment]
Looks like https://github.com/JuliaPackaging/BinaryBuilderBase.jl/blob/475288378ad6aa99bb3eeffe7ec67b4c1505e29a/src/Prefix.jl#L728 and https://github.com/JuliaPackaging/BinaryBuilderBase.jl/blob/475288378ad6aa99bb3eeffe7ec67b4c1505e29a/src/Prefix.jl#L201

**Comment by amontoison** [comment]
Yes @fredrikekre, It will help to compile some packages that can't be precompiled with Yggdrasil because of license issues.
For example in HSL.jl, linear solvers can't be distributed and I compile them with `run(...)` commands: https://github.com/JuliaSmoothOptimizers/HSL.jl/blob/main/deps/build.jl#L116-L130
